src/Windows/Avalonia.Win32.Automation/Interop/Automation/ISelectionProvider.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Avalonia.Win32.Automation.Marshalling;
+
+namespace Avalonia.Win32.Interop.Automation
+{
+#if NET8_0_OR_GREATER
+    [GeneratedComInterface]
+#else
+    [ComImport()]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+#endif
+    [Guid("fb8b03af-3bdf-48d4-bd36-1a65793be168")]
+    internal partial interface ISelectionProvider
+    {
+#if NET8_0_OR_GREATER
+        [return: MarshalUsing(typeof(SafeArrayMarshaller<IRawElementProviderSimple>))]
+#endif
+        IReadOnlyList<IRawElementProviderSimple> GetSelection();
```

Comment:
Why do we need to change from the array here? Are you sure that the legacy COM interop will properly handle all types implementing IReadOnlyList?

Reply:
It wasn't clear how I can implement simple array marshaller with new com interop. So, I went with IReadOnlyList instead, generic types are easier to marshal there.
I probably will spend a bit more time on trying to make it work with arrays.

Reply:
Just need to check if it works OK with the legacy interop.

Interfaces aren't public, so it should be safe, I guess